### PR TITLE
refactor: remove switch condition

### PR DESCRIPTION
**Pull request by Link-the-elf**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/zone.js/lib/common/utils.ts [hunk 1]
```diff
@@ -289,25 +289,12 @@ export function patchClass(className: string) {
...
-      case 3:
-        this[originalInstanceKey] = new OriginalClass(a[0], a[1], a[2]);
-        break;
-      case 4:
-        this[originalInstanceKey] = new OriginalClass(a[0], a[1], a[2], a[3]);
-        break;
-      default:
-        throw new Error('Arg list too long.');
+
+    if (a.length > 4) {
```
Comment (alfaproject) [inline packages/zone.js/lib/common/utils.ts:293]: Is this even needed now?

Comment (Link-the-elf) [reply packages/zone.js/lib/common/utils.ts:293]: @alfaproject Why not? Clear code -> hight perfomance

Comment (alfaproject) [reply packages/zone.js/lib/common/utils.ts:293]: I mean, the number of arguments shouldn't matter anymore. So this condition/error shouldn't be needed anymore

**Comment by JeanMeche** [comment]
You didn't sign the CLA so I'll close this issue. If you'd like to land this change, feel free to reopen a PR.

**Pull request closed without merging**